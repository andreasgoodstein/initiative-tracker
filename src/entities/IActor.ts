export default interface IActor {
  readonly id: number;
  readonly name: string;

  initiative?: number;

  hasActiveTurn: boolean;

  clone(id: number | undefined): IActor;
  sort(a: IActor, b: IActor): number;
}

export class Actor implements IActor {
  public readonly id: number;
  public readonly name: string;

  public hasActiveTurn: boolean;

  public initiative?: number;

  constructor(id: number, name: string) {
    this.id = id;
    this.name = name;

    this.hasActiveTurn = false;
  }

  public clone(id: number | undefined): Actor {
    return new Actor(id || this.id, this.name);
  }

  public sort(a: Actor, b: Actor): number {
    return (b.initiative || 0) - (a.initiative || 0);
  }
}
