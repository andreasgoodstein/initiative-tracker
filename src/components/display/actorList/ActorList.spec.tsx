import { shallow } from "enzyme";
import React from "react";

import IActor from "entities/IActor";

import ActorList from "./index";

const testList: IActor[] = [
  {
    hasActiveTurn: false,
    id: 1,
    initiative: 30,
    name: "Test Actor"
  }
];

describe("the ActorList component", () => {
  it("renders", () => {
    const wrapper = shallow(
      <ActorList actorList={[...testList]} removeActor={() => undefined} />
    );

    expect(wrapper).not.toBeNull();
    expect(wrapper).toMatchSnapshot();
  });
});
