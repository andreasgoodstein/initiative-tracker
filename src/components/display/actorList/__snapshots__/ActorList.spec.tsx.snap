// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ActorList component renders 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <ScrollViewMock
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <TouchableHighlight
      activeOpacity={0.85}
      delayPressOut={100}
      key="1"
      onPress={[Function]}
      underlayColor="black"
    >
      <View>
        <Component
          actor={
            Object {
              "clone": [Function],
              "hasActiveTurn": false,
              "id": 1,
              "initiative": 30,
              "name": "Test Actor",
              "sort": [Function],
            }
          }
        />
      </View>
    </TouchableHighlight>
  </ScrollViewMock>
</View>
`;
